imap/http_caldav.c

```diff
@@ -8029,14 +8051,26 @@ static int report_fb_query(struct transaction_t *txn,
                     fbfilter.end = icaltime_from_string((char *) end);
                     xmlFree(end);
                 }
-
-                if (!is_valid_timerange(fbfilter.start, fbfilter.end)) {
-                    return HTTP_BAD_REQUEST;
-                }
             }
         }
     }
 
+    if (icaltime_is_null_time(fbfilter.start)) {
+        if (icaltime_is_null_time(fbfilter.end))
+            return HTTP_BAD_REQUEST;
+
+        fbfilter.start =
+            icaltime_from_timet_with_zone(caldav_epoch, 0, utc_zone);
+    }
+    else if (icaltime_is_null_time(fbfilter.end)) {
+        fbfilter.end =
+            icaltime_from_timet_with_zone(caldav_eternity, 0, utc_zone);
+    }
+
+    if (!is_valid_timerange(fbfilter.start, fbfilter.end)) {
```

Comment:
I think here `else if` can be used - `is_valid_timerange(fbfilter.start, fbfilter.end)` is executed only if the `time-range` filter contained both `start` and `end`.